src/videojs-hls.js

```diff
@@ -320,10 +324,35 @@ videojs.Hls.prototype.setupSourceBuffer_ = function() {
   // transition the sourcebuffer to the ended state if we've hit the end of
   // the playlist
   this.sourceBuffer.addEventListener('updateend', function() {
+    var segmentInfo = this.pendingSegment_, i, currentBuffered;
+
+    this.pendingSegment_ = null;
+
     if (this.duration() !== Infinity &&
         this.mediaIndex === this.playlists.media().segments.length) {
       this.mediaSource.endOfStream();
     }
+
+    // When switching renditions or seeking, we may misjudge the media
+    // index to request to continue playback. Check after each append
+    // that a gap hasn't appeared in the buffered region and adjust
+    // the media index to fill it if necessary
+    if (this.tech_.buffered().length === 2 &&
+        segmentInfo.playlist === this.playlists.media()) {
+      i = this.tech_.buffered().length;
+      while (i--) {
+        if (this.tech_.currentTime() < this.tech_.buffered().start(i)) {
```

Comment:
This doesn't handle the case where the mediaIndex is for a segment that is too early in the timeline, correct. I know that case will be handled "automatically" but it feels suboptimal and I don't think we ever correct the "TimelineOffset" in that case which might, in the long run, be a confusing behavior.

In other words, I think we should always attempt to correct the timeline offset because some other code may depend on that correction one day.